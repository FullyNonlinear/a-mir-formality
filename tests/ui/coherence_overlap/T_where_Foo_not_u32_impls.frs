// Test positive impl that has a where-clause which checks for itself,
// i.e., `T: Foo where T: Foo`. This `T: Foo` where-clause isn't harmful
// in the coinductive interpretation of trait matching, it actually
// doesn't change the meaning of the impl at all. However, this formulation
// was erroneously accepted by an earlier variant of negative impls.
[
    crate core {
        trait Foo {}
        impl<ty T> Foo for T where T: Foo {}
        impl !Foo for u32 {}
    }
]
