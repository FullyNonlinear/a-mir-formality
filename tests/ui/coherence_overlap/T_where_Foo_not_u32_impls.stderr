Error: check_trait_impl(impl <ty> Foo for ^ty0_0 where ^ty0_0 : Foo { })

Caused by:
    failed to disprove
        {! Foo(!ty_1)}
    given
        {Foo(!ty_1)}
    got
    {Constraints { env: Env { variables: [?ty_1], coherence_mode: false }, known_true: true, substitution: {?ty_1 => u32} }}
